mmcv/runner/iter_based_runner.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (c) Open-MMLab. All rights reserved.
+import os.path as osp
+import time
+
+import torch
+from torch.optim import Optimizer
+
+import mmcv
+from .base_runner import BaseRunner
+from .checkpoint import save_checkpoint
+from .utils import get_host_info
+
+
+class IterLoader(object):
+
+    def __init__(self, dataloader):
+        self._dataloader = dataloader
+        self.iter_loader = iter(self._dataloader)
+        self._epoch = 0
+
+    @property
+    def epoch(self):
+        return self._epoch
+
+    def __next__(self):
+        try:
+            data = next(self.iter_loader)
+        except StopIteration:
+            self._epoch += 1
+            if hasattr(self._dataloader.sampler, 'set_epoch'):
+                self._dataloader.sampler.set_epoch(self._epoch)
+            self.iter_loader = iter(self._dataloader)
+            data = next(self.iter_loader)
+
+        return data
+
+    def __len__(self):
+        return len(self._dataloader)
```

Comment:
What's the meaning or usage of `__len__` of `IterLoader`? 
It could be confusing since IterLoader will re-iter once hit the end.

Reply:
Since `torch.utils.data.DataLoader` has the `__len__` api, somewhere in the code that uses `IterLoader` to replace `torch.utils.data.DataLoader` would call this method. Thus, we expose this api.